Feature/add new service

Daniel-NudgeDigital opened the pull request:
**description**

add functionality to allow a user to add a new service 

**test instruction**

- pull this branch 
- `yarn install`
- navigate to **/services/add**
- navigate through each page of the form and check functionality works and is consistent with designs 
- check hidden fields appear as appropriate (as shown on figma designs) 
- pay particular attention to validation logic on page 2 (service locations page) i.e. you must enter a location, you can enter multiple locations, locations cant be duplicate, you can add a location and change postcode and/or remove location as many times as you like without breaking the validation or getting any console errors 
- you can add and remove an image on page 5 (service image page)

figma designs: 

https://www.figma.com/file/t6fanDKvLkCHJsY85ivFTF/Services-Directory?node-id=586%3A8553

![image](https://user-images.githubusercontent.com/69238887/94475639-47397680-01c7-11eb-97f1-074ba5120f3f.png)


**ticket**

https://app.clickup.com/t/9e5u7e
https://app.clickup.com/t/9e5u7j
https://app.clickup.com/t/9e5u7q
https://app.clickup.com/t/78ntbf
https://app.clickup.com/t/78ntch
https://app.clickup.com/t/9e5u7g
https://app.clickup.com/t/9e5u7n
https://app.clickup.com/t/9e5u7c
https://app.clickup.com/t/9e5u7x
https://app.clickup.com/t/9e5u7u



**note**
if you skip back to page 2 using the menu navigation the postcode data will not be displayed - this will be picked up on a subsequent commit

samwongnudge commented:
Hey @Daniel-NudgeDigital great work so far. Let me know what you think of the bits below

Step 1
* Styling
  * consider reducing the spacing between the organisation name and menu to be more aligned to the designs
  * Tab menu labels aren't the same as design, but I don't know how important this is
  * Strangely the designs have a white page background and a border around only the main content element. I think I prefer what you have except maybe the line under the last item in the sidebar nav

Step 2
* Functionality
  * Maybe instead of "Please enter a location", have a more descriptive message explaining why. I think the client said they need some other text to explain this / and contact if they don't have a location so they can use the town hall location or alike
  * What will the "I can't find my address in the list" do?
  * Do you want to display address2 in the select address?
  * Try adding 2 complete locations (so 2 markers are shown). Then click 'change' the first postcode, specifically this one doesn't have a remove button and the first item becomes blank and the 2nd is filled. Then click continue. In terms of the data you're passing through, is this okay?
  * Going back from step 3 didn't store state of my addresses, is this correct?
* Styling
  * Be nice if you could override the default icons, they look very basic. I can help you out with this if you need to know how
  * Can the continue button be moved after the map?
  * The map also has a 'Map preview' heading too which would be nice context of what it's for
  * Will you also be using the client mapbox credentials? Should be right? You'll need to change the map url at some point and have the refer to a token in a gitignored file (example and readme updates would accompany this)

Step 3
* Functionality
  * Each checkbox needs a hidden field so a custom description can be provided for any of the selected categories
  * I think this should be mandatory - a category must be assigned
* Styling
  * The 2nd sentence doesn't really make sense. Adding 'For' at the start would improve it

Step 4
* Functionality
  * If I don't select anything here, should it select 'everyone' by default? And also should it be selected if I go back?
  * It's designed like this, but I don't really like this functionality. It should have 'everyone' as the first item in the list, clicking this should check all following items underneath and vice versa - I think this is the better UX, but keep as is for now
* Styling
  * Remove the 'Search keywords' and paragraph underneath, probably added in by error

Step 5
* Functionality
  * Would it be helpful to ask the user to add an image of a particular rotation / ratio? They'll want their image to come out looking correct right, so you need the user to first pass in the correct image before passing it to the backend to process to render

Daniel-NudgeDigital commented:
Thanks @samwongnudge for comments, really helpful 

Below are responses to comments (any comments not included below have been actioned as per your recommendation) 

> 
> Step 2
> 
> Maybe instead of "Please enter a location", have a more descriptive message explaining why. I think the client said they need some other text to explain this / and contact if they don't have a location so they can use the town hall location or alike

Probably best to add further text at the top of the page, i'll ask hackney team for copy 

>   * What will the "I can't find my address in the list" do?

I suspect this will link to an external page, I'll check with hackney team 

>   * Try adding 2 complete locations (so 2 markers are shown). Then click 'change' the first postcode, specifically this one doesn't have a remove button and the first item becomes blank and the 2nd is filled. Then click continue. In terms of the data you're passing through, is this okay?

Yes this is handled prior to sending results to the API 

>   * Going back from step 3 didn't store state of my addresses, is this correct?

Thanks for pointing this out, I am aware of this behaviour and will address it in a subsequent MR 

>   * Be nice if you could override the default icons, they look very basic. I can help you out with this if you need to know how

Agree this would be good to do, I've created a separate task for this in ClickUp [https://app.clickup.com/t/8mv4a8](https://app.clickup.com/t/8mv4a8)


>   * Will you also be using the client mapbox credentials? Should be right? You'll need to change the map url at some point and have the refer to a token in a gitignored file (example and readme updates would accompany this)

Thanks for picking up on this, i've now updated to use mapbox credentials 

> Step 4
> 
> * Functionality
>   
>   * If I don't select anything here, should it select 'everyone' by default? And also should it be selected if I go back?
>   * It's designed like this, but I don't really like this functionality. It should have 'everyone' as the first item in the list, clicking this should check all following items underneath and vice versa - I think this is the better UX, but keep as is for now

Agree with your suggestion and have now updated 

> Step 5
> 
> * Functionality
>   
>   * Would it be helpful to ask the user to add an image of a particular rotation / ratio? They'll want their image to come out looking correct right, so you need the user to first pass in the correct image before passing it to the backend to process to render

Is there a particular rotation/ratio the public frontend is expecting where these images will be presented? or something i should pickup with martin?

samwongnudge commented:
Hey @Daniel-NudgeDigital great stuff mate, well done on making the changes - all looks good and agree with what you've said / done

Image ratio: Designs don't give an exact one, but if keeping it simple - roughly around landscape 2:1 seems around right
